bin/03_yolo_opencv.py

```diff
@@ -3,141 +3,44 @@
-        x ([type]): [description]
-        y ([type]): [description]
-        x_plus_w ([type]): [description]
-        y_plus_h ([type]): [description]
-    """
-    label = str(classes[class_id])
-    color = COLORS[class_id]
-    cv2.rectangle(img, (x, y), (x_plus_w, y_plus_h), color, 2)
-    cv2.putText(img, label, (x - 10, y - 10), cv2.FONT_HERSHEY_SIMPLEX, 0.5, color, 2)
-
-
-# run inference through the network
-# and gather predictions from output layers
-outs = net.forward(get_output_layers(net))
-
-# initialization
-class_ids = []
-confidences = []
-boxes = []
-conf_threshold = 0.3
-nms_threshold = 0.4
-
-# for each detetion from each output layer
-# get the confidence, class id, bounding box params
-# and ignore weak detections (confidence < 0.5)
-for out in outs:
-    for detection in out:
-        scores = detection[5:]
-        class_id = np.argmax(scores)
-        confidence = scores[class_id]
-        if confidence > 0.5:
-            center_x = int(detection[0] * Width)
-            center_y = int(detection[1] * Height)
-            w = int(detection[2] * Width)
-            h = int(detection[3] * Height)
-            x = center_x - w / 2
-            y = center_y - h / 2
-            class_ids.append(class_id)
-            confidences.append(float(confidence))
-            boxes.append([x, y, w, h])
-
-# apply non-max suppression
-indices = cv2.dnn.NMSBoxes(boxes, confidences, conf_threshold, nms_threshold)
-
-# go through the detections remaining
-# after nms and draw bounding box
-person_count = 0
-
-for i in indices.flatten():
-    # draw for persons only
-    if class_ids[i] != 0:
-        continue
-    # increment person count
-    person_count += 1
-    # draw bounding box
-    box = boxes[i]
-    x = box[0]
-    y = box[1]
-    w = box[2]
-    h = box[3]
-    draw_bounding_box(
-        image,
-        class_ids[i],
-        confidences[i],
-        round(x),
-        round(y),
-        round(x + w),
-        round(y + h),
-    )
-
-print(f"Number on persons: {person_count}")
-
-# save image
-plt.axis("off")
-plt.imshow(
-    cv2.cvtColor(image, cv2.COLOR_BGR2RGB), interpolation="bicubic", aspect="auto"
-)
-plt.savefig(paths.OUTPUT_DIR / image_path.name)
+from traffic_cam import predictor
+
+
+if __name__ == "__main__":
+    try:
+        predictor = predictor.Predictor()
+
+        images = {
+            "image_2020-09-09T20:13:29.985893+00:00s": 1,
+            "image_2020-09-10T07:58:06.167747+00:00": 4,
+            "image_2020-09-25T12:51:31.666045+00:00": 15,
+            "image_2020-09-25T12:51:52.302745+00:00": 23,
+            "image_2020-09-25T12:52:08.598229+00:00": 24,
+            "image_2020-09-25T12:55:24.172202+00:00": 23,
+            "image_2020-09-25T13:59:54.823420+00:00": 34,
+            "image_2020-09-25T14:00:11.859519+00:00": 22,
+            "image_2020-09-25T14:00:28.696451+00:00": 25,
+            "image_2020-09-25T14:03:33.700749+00:00": 29,
+        }
+
+        squares = []
```

Comment:
the higher-level purpose of this error calculation is not obvious -> maybe add high-level comments and a more verbose print in the end.
on a lower level, the variables `squares`, `prediction`, `error` and `nrmsd` could have a more self-explanatory naming, I think.

overall, I very much like the to benchmark the model this way.
maybe an MAE error metric would be easier for end-users?

Reply:
btw, changing the metric is highly optional. maybe even out of scope, for now.

Reply:
See #20